x-pack/plugins/secops/public/components/formatted_duration/helpers.test.ts

```diff
@@ -0,0 +1,401 @@
+        '23h 59m 59s 999.999999ms'
+      );
+    });
+
+    test('it returns the duration formatted as 1 day, 0 hours, 0 minutes, and 0 seconds, (and no milliseconds) when the duration is exactly one day', () => {
+      expect(getFormattedDurationString(ONE_DAY * ONE_MILLISECOND_AS_NANOSECONDS)).toEqual(
+        '1d 0h 0m 0s'
+      );
+    });
+
+    test('it returns the duration formatted as one day, with zero hours, minutes, seconds, and fractional milliseconds when the duration is one day + 1 nanosecond', () => {
+      expect(getFormattedDurationString(ONE_DAY * ONE_MILLISECOND_AS_NANOSECONDS + 1)).toEqual(
+        '1d 0h 0m 0s 0.000001ms'
+      );
+    });
+
+    test('it returns the duration formatted as 29 days, 23 hours, 59 minutes, 59 seconds, and with fractional milliseconds when the duration is one month - 1 nanosecond', () => {
+      expect(getFormattedDurationString(ONE_MONTH * ONE_MILLISECOND_AS_NANOSECONDS - 1)).toEqual(
+        '29d 23h 59m 59s 999.999999ms'
+      );
+    });
+
+    test('it returns 30 days, zero hours, minutes, seconds, (and no millieconds) when the duration is exactly one month, as is the current behavior of moment', () => {
+      expect(getFormattedDurationString(ONE_MONTH * ONE_MILLISECOND_AS_NANOSECONDS)).toEqual(
+        '30d 0h 0m 0s' // see https://github.com/moment/moment/issues/3653
+      );
+    });
+
+    test('it returns the duration as 29 days, 23 hours, 59 minutes, 59 seconds, and fractional milliseconds when the duration is 1 month - 1 nanosecond', () => {
+      expect(getFormattedDurationString(ONE_MONTH * ONE_MILLISECOND_AS_NANOSECONDS - 1)).toEqual(
+        '29d 23h 59m 59s 999.999999ms' // see https://github.com/moment/moment/issues/3653
+      );
+    });
+
+    test('it returns 1 month, zero days, hours, minutes, seconds (and no milliseconds) month when the duration is exactly 1 month + 1 day, as is the current behavior of moment', () => {
+      expect(
+        getFormattedDurationString((ONE_MONTH + ONE_DAY) * ONE_MILLISECOND_AS_NANOSECONDS)
+      ).toEqual(
+        '1m 0d 0h 0m 0s' // see https://github.com/moment/moment/issues/3653
+      );
+    });
+
+    test('it returns the 1 month with 0 days, hours, minutes, seconds, and fractional milliseconds when the duration is exactly 1 month + 1 day + 1 nanosecond, ', () => {
+      expect(
+        getFormattedDurationString((ONE_MONTH + ONE_DAY) * ONE_MILLISECOND_AS_NANOSECONDS + 1)
+      ).toEqual(
+        '1m 0d 0h 0m 0s 0.000001ms' // see https://github.com/moment/moment/issues/3653
+      );
+    });
+
+    test('it returns 11 months, 30 days (with 0 hours, minutes, and non-fractional seconds) when the duration is exactly one year, as is the current behavior of moment', () => {
+      expect(getFormattedDurationString(ONE_YEAR * ONE_MILLISECOND_AS_NANOSECONDS)).toEqual(
+        '11m 30d 0h 0m 0s' // see https://github.com/moment/moment/issues/3209
```

Comment:
Thanks for the links to the issue tracking so we know when to remove this when it is fixed.